pkg/build/controller/strategy/custom.go

```diff
@@ -17,14 +17,14 @@ type CustomBuildStrategy struct {
 
 // CreateBuildPod creates the pod to be used for the Custom build
 func (bs *CustomBuildStrategy) CreateBuildPod(build *buildapi.Build) (*kapi.Pod, error) {
-	buildJSON, err := json.Marshal(build)
+	data, err := v1beta1.Codec.Encode(build)
```

Comment:
Inject the codec from the BuildControllerFactory through the typeBasedFactoryStrategy.